lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -1811,24 +1832,16 @@ public function PartialObjectExpression()
     }
 
     /**
-     * NewObjectExpression ::= "NEW" IdentificationVariable "(" NewObjectArg {"," NewObjectArg}* ")"
+     * NewObjectExpression ::= "NEW" AbstractSchemaName "(" NewObjectArg {"," NewObjectArg}* ")"
      *
      * @return \Doctrine\ORM\Query\AST\NewObjectExpression
      */
     public function NewObjectExpression()
     {
         $this->match(Lexer::T_NEW);
-        $this->match(Lexer::T_IDENTIFIER);
-
-        $token      = $this->lexer->token;
-        $className  = $token['value'];
-
-        if (strrpos($className, ':') !== false) {
-            list($namespaceAlias, $simpleClassName) = explode(':', $className);
 
-            $className = $this->em->getConfiguration()
-                ->getEntityNamespace($namespaceAlias) . '\\' . $simpleClassName;
-        }
+        $className = $this->AbstractSchemaName(); // note that this is not yet validated
```

Comment:
I don't know. It was like this before, only that I switched from expecting an `identifier` token here to an `AbstractSchemaName` (and I had to decouple the latter from the validation to keep things as they were previously).